nncf/quantization/passes.py

```diff
@@ -109,15 +116,27 @@ def filter_constant_nodes(
     if not start_nodes:
         return nncf_graph
 
-    visited_nodes = set()
-    nodes_queue = collections.deque(start_nodes)
-    while nodes_queue:
-        node = nodes_queue.pop()
-        if node in visited_nodes:
-            continue
-        visited_nodes.add(node)
-        nodes_queue.extend(nncf_graph.get_next_nodes(node))
-    constant_nodes = [node for node in nncf_graph.get_all_nodes() if node not in visited_nodes]
+    def fill_nodes_list(initial_list: List[NNCFNode], fill_list: Set[NNCFNode], extend_call: callable):
```

Comment:
```suggestion
    def visit_nodes(start_nodes: List[NNCFNode], visited_nodes: Set[NNCFNode], extend_call: Callable[NNCFNode, List[NNCFNode]]):
```

Reply:
Don't see any improvement in suggestions.

Reply:
Take a look at the typehints

Reply:
Done